File: packages/cli/src/mascot/MascotState.ts
/** Full 10-state vocabulary the daemon may send (ARCHITECTURE.md §13). */
export type MascotStateName =
	| "thinking"
	| "coding"
	| "analyzing"
	| "planning"
	| "building"
	| "terminal"
	| "tool"
	| "waiting"
	| "done"
	| "presenting";

/** What the CLI actually knows how to render as a distinct animation this phase. */
export type RenderedMascotState = "thinking" | "coding" | "idle";

const RENDERED_STATES = new Set<RenderedMascotState>(["thinking", "coding"]);

/**
 * Only "thinking" and "coding" get real animations in Phase 1. Every other state in the
 * 10-state vocabulary (analyzing/planning/building/terminal/tool/waiting/done/presenting)
 * must not crash the client — it just falls back to a neutral idle look.
 */
export function toRenderedState(state: string): RenderedMascotState {
	return RENDERED_STATES.has(state as RenderedMascotState) ? (state as RenderedMascotState) : "idle";
}

File: apps/desktop/lib/mascot/mascot_state.dart
/// The 10 mascot states from ARCHITECTURE.md §13 / OTOCLAW_PLAN.md §3, plus
/// an `idle` fallback for the pre-run / unknown-state case.
enum MascotStateName {
  idle,
  thinking,
  coding,
  analyzing,
  planning,
  building,
  terminal,
  tool,
  waiting,
  done,
  presenting,
}

/// Maps the wire `state` string (from `MascotStatePayload.state`) to a
/// [MascotStateName]. Unknown strings fall back to [MascotStateName.idle].
MascotStateName mascotStateFromWire(String state) {
  switch (state) {
    case 'thinking':
      return MascotStateName.thinking;
    case 'coding':
      return MascotStateName.coding;
    case 'analyzing':
      return MascotStateName.analyzing;
    case 'planning':
      return MascotStateName.planning;
    case 'building':
      return MascotStateName.building;
    case 'terminal':
      return MascotStateName.terminal;
    case 'tool':
      return MascotStateName.tool;
    case 'waiting':
      return MascotStateName.waiting;
    case 'done':
      return MascotStateName.done;
    case 'presenting':
      return MascotStateName.presenting;
    default:
      return MascotStateName.idle;
  }
}
